src/Employer/Employer.Web/Controllers/ApplicationReviewController.cs

```diff
@@ -18,25 +18,47 @@ public ApplicationReviewController(ApplicationReviewOrchestrator orchestrator)
             _orchestrator = orchestrator;
         }
 
-        [HttpGet("", Name = RouteNames.ApplicationReview_Get)]
+        [HttpGet("review", Name = RouteNames.ApplicationReview_Get)]
         public async Task<IActionResult> ApplicationReview(ApplicationReviewRouteModel rm)
         {
             var vm = await _orchestrator.GetApplicationReviewViewModelAsync(rm);
             return View(vm);
         }
 
-        [HttpPost("", Name = RouteNames.ApplicationReview_Post)]
+        [HttpPost("review", Name = RouteNames.ApplicationReview_Post)]
         public async Task<IActionResult> ApplicationReview(ApplicationReviewEditModel m)
         {
-            if (ModelState.IsValid == false)
+            if (!ModelState.IsValid)
             {
                 var vm = await _orchestrator.GetApplicationReviewViewModelAsync(m);
                 return View(vm);
-            }
+            }            
+            var applicationStatusConfirmationViewModel = await _orchestrator.GetApplicationStatusConfirmationViewModelAsync(m);
+            return View("ApplicationStatusConfirmation", applicationStatusConfirmationViewModel);
+        }
 
-            await _orchestrator.PostApplicationReviewEditModelAsync(m, User.ToVacancyUser());
+       [HttpPost("status", Name = RouteNames.ApplicationStatusConfirmation_Post)]
+        public async Task<IActionResult> ApplicationStatusConfirmation(ApplicationReviewStatusConfirmationEditModel applicationReviewStatusConfirmationEditModel)
+        {            
+            if (!ModelState.IsValid)
+            {
+                var vm = await _orchestrator.GetApplicationStatusConfirmationViewModelAsync(applicationReviewStatusConfirmationEditModel);
+                return View(vm);
+            }
 
-            return RedirectToRoute(RouteNames.VacancyManage_Get);
-        }
+            if (applicationReviewStatusConfirmationEditModel.NotifyApplicant.HasValue &&
+                applicationReviewStatusConfirmationEditModel.NotifyApplicant.Value)
+            {
+                await _orchestrator.PostApplicationReviewEditModelAsync(applicationReviewStatusConfirmationEditModel, User.ToVacancyUser());
+                var routeModel=new ApplicationReviewRouteModel {
+                    ApplicationReviewId = applicationReviewStatusConfirmationEditModel.ApplicationReviewId,
+                    VacancyId = applicationReviewStatusConfirmationEditModel.VacancyId,
+                    EmployerAccountId = applicationReviewStatusConfirmationEditModel.EmployerAccountId
+                };
+                var vm = await _orchestrator.GetApplicationReviewViewModelAsync(routeModel);
+                return RedirectToRoute(RouteNames.VacancyManage_Get, new { applicationUserName =vm.Name, applicationReviewStatus = applicationReviewStatusConfirmationEditModel.Outcome.ToString(), setApplicationStatus ="true"});
```

Comment:
We use TempData to display messages when redirecting to other pages. Have a look at `TempDataKeys.DashboardInfoMessage` to see how this works. This should simplify this function somewhat.